lab-anthony/test/drink-router-success-test.js

```diff
@@ -0,0 +1,146 @@
+const Drink = require('../model/drink');
+const port = process.env.PORT || 3000;
+const baseUrl = `localhost:${port}/api/drink`;
+
+describe('testing module drink-router', function(){
+  before((done)=>{
+    if(!server.isRunning){
+      server.listen(port, ()=>{
+        console.log('server running on port', port);
+        done();
+      });
+      return;
+    }
+    done();
+  });
+  after((done)=>{
+    if (server.isRunning){
+      server.close(()=>{
+        server.isRunning = false;
+        console.log('server shutdown');
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  describe('testing POST /api/drink', function(){
+    after((done)=>{
+      storage.pool = {};
+      done();
+    });
+
+    it('should return a drink', function(done){
+      request.post(baseUrl)
+      .send({drinkName:'Macchiato', cafeName:'Cafe Javasti'})
+      .end((err, res)=>{
+        expect(res.status).to.equal(200);
+        expect(res.body.drinkName).to.equal('Macchiato');
+        expect(res.body.cafeName).to.equal('Cafe Javasti');
+        done();
+      });
+    });
+  });
+
+  describe('testing GET /api/drink', function(){
+    before((done)=>{
+      this.tempDrink = new Drink('Macchiato', 'Cafe Javasti');
+      storage.setItem('drink', this.tempDrink);
+      done();
+    });
+
+    after((done)=>{
+      storage.pool = {};
+      done();
+    });
+
+    it('should return a drink', (done)=>{
+      request.get(`${baseUrl}/${this.tempDrink.id}`)
+        .end((err, res)=>{
+          expect(res.status).to.equal(200);
+          expect(res.body.drinkName).to.equal('Macchiato');
+          expect(res.body.cafeName).to.equal('Cafe Javasti');
+          done();
+        });
+    });
+  });
+
+  describe('testing PUT /api/drink', function(){
+    before((done)=>{
+      this.tempDrink = new Drink('Macchiato', 'Cafe Javasti');
+      storage.setItem('drink', this.tempDrink);
+      done();
+    });
+    after((done)=>{
+      storage.pool = {};
+      done();
+    });
+
+    it('should return a drink with an updated name "Mocha"', (done)=>{
+      request.put(`${baseUrl}/${this.tempDrink.id}`)
+        .send({drinkName: 'Mocha'})
+        .end((err, res)=>{
+          expect(res.status).to.equal(200);
+          expect(res.body.drinkName).to.equal('Mocha');
+          expect(res.body.cafeName).to.equal('Cafe Javasti');
+          done();
+        });
+    });
+  });
+
+  describe('testing DELETE /api/drink', function(){
+    before((done)=>{
+      this.tempDrink = new Drink('Macchiato', 'Cafe Javasti');
+      storage.setItem('drink', this.tempDrink);
+      done();
+    });
+
+    after((done)=>{
+      storage.pool = {};
+      done();
+    });
+
+    it('should return successful delete as true', (done)=>{
+      request.delete(`${baseUrl}/${this.tempDrink.id}`)
+        .end((err, res)=>{
+          expect(res.status).to.equal(200);
+          expect(res.body.drinkName).to.equal(undefined);
```

Comment:
This is a little weird and seems like syntax that could throw an error. 

That said, good work writing the delete tests!